src/main/java/org/springframework/retry/support/RetryTemplateBuilder.java

```diff
@@ -132,6 +133,20 @@ public RetryTemplateBuilder withinMillis(long timeout) {
 		return this;
 	}
 
+	/**
+	 * Retry until {@code timeout} has passed since the initial attempt.
+	 * @param timeout duration for how long retries should be attempted
+	 * @return this
+	 * @see TimeoutRetryPolicy
+	 * @throws IllegalArgumentException if timeout is {@code null} or 0, or if another
+	 * retry policy has already been configured.
+	 * @since 2.0.2
+	 */
+	public RetryTemplateBuilder withTimeout(Duration timeout) {
```

Comment:
I found `withinMillis()` not to be suitable as a method name if the argument is a `java.time.Duration`. Therefore, I picked `withTimeout()`.

I suggest to deprecate `withinMillis` and to add a `withTimeout(long)` instead. Can do this in a follow-up PR.

Reply:
Sure! What you are saying totally makes sense and you are free to do that deprecation directly in this PR.
Also, please, add your name to the `@author` of the affected classes.

Reply:
Is it acceptable to deprecate a method in a patch release? It might break the build for people that treat warnings as errors. Might be less disruptive to do that in a 2.1.0. My question would then be how to proceed. Change `@since` to 2.1.0?

Reply:
Yes, it is acceptable.
If you upgrade and your build fails for those deprecations, then you'd just need to fix them.
Luckily we will have an easy pass to migrate.
Then in the next version we might just simply to remove this deprecated method.

I my feeling it is not OK to treat a deprecation warning as an error.